public/js/widget_config.js

```diff
@@ -198,20 +203,40 @@ const openConfigureWidgetDialog = function (widget) {
         jQuery(this).dialog('close')
       },
       Done: function () {
-        reconfigureWidget(widgetConfig)
+        reconfigureWidget(oldWidgetConfig, extractWidgetConfigurationFromDialog())
         jQuery(this).dialog('close')
       }
     }
   }).dialog('open')
 }
 
-const reconfigureWidget = function (widgetConfig) {
-  console.log("reconfiguring widgets hasn't been implemented yet")
+/*
+* @returns {object} - the dragable jQuery widget element
+* TODO: We really need a better way of identifying widgets, not using their labels.
+*/
+const getjQueryWidgetFromConfig = function (widgetConfig) {
+  return jQuery('#' + widgetConfig.label)
+}
+
+/*
+* Simply delete the widget and re-create it with the new config settings
+* TODO: In the future, look at attempting to call a "update" method on the widget before defaulting to this method
+*/
+const reconfigureWidget = function (oldWidgetConfig, newWidgetConfig) {
+  const oldDragableWidget = getjQueryWidgetFromConfig(oldWidgetConfig)
+
+  oldDragableWidget.remove()
+  positionWidgets()
+
+  newWidgetConfig.position = oldWidgetConfig.position
+  createWidget(newWidgetConfig)
+  positionWidgets()
 }
 
 const setNewWidgetDialogType = function (widgetType) {
   jQuery('#newWidget .newWidgetType').hide()
   jQuery(`#newWidget #${widgetType}`).show()
+  positionWidgets()
```

Comment:
Without line 239, the widgets would all get jumbled when changing the widget type